service/src/main/java/org/exoplatform/extension/exchange/service/util/CalendarConverterUtils.java

```diff
@@ -184,7 +185,8 @@ public static void convertExchangeToExoMasterRecurringCalendarEvent(CalendarEven
           }
         }
         event.setRepeatByDay(daysList.toArray(new String[0]));
-        Utils.adaptRepeatRule(event, DEFAULT_TIME_ZONE , CalendarService.PERSISTED_TIMEZONE);
+        CalendarSetting calendarSetting = getCalendarService().getCalendarSetting(username);
+        Utils.adaptRepeatRule(event, DateUtils.getTimeZone(calendarSetting.getTimeZone()), CalendarService.PERSISTED_TIMEZONE);

```

Comment:
please split this instruction to multiple ones, like follows:

```java
String userTimeZoneId = null;
if (calendarSetting == null) {
    if(LOG.isDebugEnabled()) {
      LOG.warn("User {} doesn't have calendar settings. Default timezone will be used.", username);
    }
    userTimeZoneId = System.defaultTimeZone();
} else {
  String userTimeZoneId = calendarSetting.getTimeZone();
  if (StringUtils.isBlank(userTimeZoneId)) {
    if(LOG.isDebugEnabled()) {
      LOG.warn("User {} doesn't have a predefined timezone. Default will be used.", username);
    }
    userTimeZoneId = System.defaultTimeZone();
  }
}
TimeZone userTimeZone = DateUtils.getTimeZone(userTimeZoneId);
if (userTimeZone == null) {
  if(LOG.isDebugEnabled()) {
    LOG.warn(("User  {} time zone settings {} isn't recognized. Default will be used.", username, userTimeZoneId);
   }
  userTimeZone = // Default time zone
}
Utils.adaptRepeatRule(event, userTimeZone, CalendarService.PERSISTED_TIMEZONE);

```

Reply:
Fixed.